### PR TITLE
fix(tailwind): parse non inline configuration variables

### PR DESCRIPTION
**Pull request by dielduarte**
<!-- Thanks for opening a PR! Your contribution is much appreciated.
To make sure your PR is handled as smoothly as possible we request that you follow the checklist sections below.
Choose the right checklist for the change(s) that you're making:

The PR name should follow `<type>(<scope>): <Message>`

Examples:
- New feature: `feat(button): Add new thing`
- Fix: `fix(react-email): Dev command
- Misc/Chore: `chore(root): Update `readme.md`
-->

## Description
getTailwindConfig only handled cases where the <Tailwind config={...}> prop was an inline object literal. If the config was stored in a variable (e.g., const config = { ... }; <Tailwind config={config} />), it failed to parse it.                                                                                              
                                                                                                                 
The Fix: When the config prop value is an Identifier (a variable reference), a new findVariableInitializer helper traverses the AST to locate that variable's declaration and extract its source code, then uses that as the config value.                                                                                                          
                                                                                                                 

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Fixes Tailwind config parsing when the `config` prop uses a variable instead of an inline object. The compatibility check now resolves the variable initializer from the right scope (even with shadowed names), addressing DEV-361.

- **Bug Fixes**
  - Resolve `config` identifiers with scope-aware AST binding and extract the initializer source; keep parsing local (avoid shared helper).
  - Add tests for variable config, empty config, and shadowed `config` names; add `biome-ignore` to allow intentional shadowing in the fixture.

<sup>Written for commit 561fdc39b3fae7755dac24e0b5383fba81c7242f. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

### CODE REVIEW

File: packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts [hunk 1]
```diff
@@ -111,6 +123,34 @@ export { reactEmailTailwindConfigInternal };`,
...
+  traverse(ast, {
+    VariableDeclarator(nodePath) {
+      if (
+        nodePath.node.id.type === 'Identifier' &&
+        nodePath.node.id.name === name &&
+        nodePath.node.init != null &&
+        nodePath.node.init.start != null &&
+        nodePath.node.init.end != null
+      ) {
+        initSource = sourceCode.slice(
```
Comment (gabrielmfern) [inline packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts]: I think the sourceCode is available in the ast isn't it?

File: packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.spec.ts [hunk 2]
```diff
@@ -4,23 +4,51 @@ import { parse } from '@babel/parser';
 import { pixelBasedPreset } from 'react-email';
 import { getTailwindConfig } from './get-tailwind-config';
 
+const parseSource = (sourceCode: string) =>
```
Comment (gabrielmfern) [inline packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.spec.ts]: do we need this function? imo it's better to have it inlined

it's easier to read that way

File: .changeset/old-shirts-tan.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"react-email": patch
+---
+
+fix(tailwind): parse non inline configuration variables
```
Comment (gabrielmfern) [inline .changeset/old-shirts-tan.md]:
```suggestion
tailwind: parse non inline configuration variables
```

Comment (gabrielmfern) [reply .changeset/old-shirts-tan.md]: changesets go into the changelog directly, so we tend to just write a description of the change instead of something like a commit message

Comment (dielduarte) [reply .changeset/old-shirts-tan.md]: done here [6a9c1f9](https://github.com/resend/react-email/pull/3466/commits/6a9c1f94c98b630713c1f2e306c3ea7baf04e7c4)

File: packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts [hunk 4]
```diff
@@ -111,6 +123,34 @@ export { reactEmailTailwindConfigInternal };`,
...
+  sourceCode: string,
+  name: string,
+): string | undefined => {
+  let initSource: string | undefined;
+
+  traverse(ast, {
+    VariableDeclarator(nodePath) {
+      if (
+        nodePath.node.id.type === 'Identifier' &&
+        nodePath.node.id.name === name &&
```
[the remark this answers is not in the document]
Comment (dielduarte) [reply packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts]: done here: [b680632](https://github.com/resend/react-email/pull/3466/commits/b6806329a2018968293298872a044ee801b92f35)

File: packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.spec.ts (hunk 2 above)
Comment (dielduarte) [reply packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.spec.ts]: no strong opinions yet, let me revert! I thought it was a bit verbose

Comment (dielduarte) [reply packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.spec.ts]: done here: [670089](https://github.com/resend/react-email/pull/3466/commits/e67008935ef290948ea4b45ff7f72cd0a1ac12b1)

File: packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts (hunk 1 above)
Comment (dielduarte) [reply packages/ui/src/utils/caniemail/tailwind/get-tailwind-config.ts]: I don't think so because AST is ReturnType<typeof parse> from @babel/parser and that's a File node. Babel's AST nodes only carry start/end position offsets

File: packages/ui/src/utils/caniemail/tailwind/tests/dummy-email-template-shadowed-config.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { pixelBasedPreset, Tailwind } from 'react-email';
+
+const _config = { wrong: true };
```
Comment (gabrielmfern) [inline packages/ui/src/utils/caniemail/tailwind/tests/dummy-email-template-shadowed-config.tsx]: this isn't shadowed though, since it's not actually the same name, right?

Comment (dielduarte) [reply packages/ui/src/utils/caniemail/tailwind/tests/dummy-email-template-shadowed-config.tsx]: trueee! lint bypasses it, fixed here: [561fdc3](https://github.com/resend/react-email/pull/3466/commits/561fdc39b3fae7755dac24e0b5383fba81c7242f)

**Review by gabrielmfern — APPROVED** [review]
